webapp/src/main/java/com/box/l10n/mojito/service/branch/notification/BranchNotificationService.java

```diff
@@ -173,8 +189,59 @@ void sendNotificationsForBranchWithSender(
       scheduleMissingScreenshotNotificationsForBranch(branch, notifierId);
     }
 
-    if (shouldSendTranslatedMessage(branch, branchNotification)) {
-      sendTranslatedMessage(branchNotificationMessageSender, branch, branchNotification);
+    handleSendBranchTranslatedMessage(branchNotificationMessageSender, branch, branchNotification);
+  }
+
+  private void handleSendBranchTranslatedMessage(
+      BranchNotificationMessageSender branchNotificationMessageSender,
+      Branch branch,
+      BranchNotification branchNotification) {
+
+    if (!shouldSendTranslatedMessage(branch, branchNotification)) {
+      // Already sent a notification or the branch is not translated yet, exit early
+      return;
+    }
+
+    Optional<BranchMergeTarget> branchMergeTargetOptional =
+        branchMergeTargetRepository.findByBranch(branch);
+    BranchStatistic branchStatistic = branchStatisticRepository.findByBranch(branch);
+
+    if (branchMergeTargetOptional.isEmpty() || !branchMergeTargetOptional.get().isTargetsMain()) {
```

Comment:
Could we encapsulate the new logic above in its own method that returns a bool so we focus here mostly on the checks?
e.g.: if(isNotTargetingMainBranch) ...